### PR TITLE
feat: Allow extension of oauth2-proxy environment variables with envFrom

### PR DESCRIPTION
**Pull request by hobti01**
## Pull request description 

Allow reference of Secret or ConfigMap to provide multiple environment variables. This allows deployers of the Helm Chart to manage multiple sensitive or non-sensitive values without needing to override or modify multiple default values.

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

- None

## Changes

- Allow extension of oauth2-proxy deployment with prexisting Secret or Configmap

## Fixes

- #611

### CODE REVIEW

**Review by vsukhin — APPROVED** [review]
thank yuo @hobti01 a lot. looks cool! for @ypoplavs to confirm

File: charts/testkube-dashboard/templates/oauth2-deployment.yaml [hunk 1]
```diff
@@ -56,52 +56,70 @@ spec:
...
+          value: "{{ .clientId }}"
           {{- end }}
+        {{- end }}
+        {{- if or .clientSecret .secretClientSecretName }}
         - name: OAUTH2_PROXY_CLIENT_SECRET
-          {{- if .Values.oauth2.env.secretClientSecretName }}
+          {{- if .secretClientSecretName }}
           valueFrom:
             secretKeyRef:
               name: {{ .Values.oauth2.env.secretClientSecretName }}
```
Comment (ypoplavs) [inline charts/testkube-dashboard/templates/oauth2-deployment.yaml:76]: I guess this should be changed to `{{ .secretClientSecretName }}` and the key to `{{ .secretClientSecretKey }}`.

File: charts/testkube-dashboard/values.yaml [hunk 2]
```diff
@@ -151,6 +151,13 @@ oauth2:
     #- --show-debug-on-error=true
   ## Array with extra environment variables
   extraEnvVars: []
+  ## Array with extra sources for environment variables
+  extraEnvFrom: []
```
Comment (ypoplavs) [inline charts/testkube-dashboard/values.yaml:155]: Could you please duplicate this var the `teskube/values.yaml` file?

Comment (hobti01) [reply charts/testkube-dashboard/values.yaml:155]: Done.  Chart versions have not been changed as I'm not sure of your policy or other locations needing a change for that.

Comment (ypoplavs) [reply charts/testkube-dashboard/values.yaml:155]: Chart version will be automatically bumped on merge to main so no need to so smth extra here.

**Pull request merged**